src/Core/mcts/include/MCTSBase.h

```diff
@@ -6,12 +6,16 @@
 #include <SearchNode.h>
 #include <State.h>
 
-class MCTSBase : MCTSInterface {
+class MCTSBase : public MCTSInterface {
   public:
     explicit MCTSBase(EnvironmentInterface &environment);
     State run(int n_searches);
+    State run(int n_searches, State initial_state);
     EnvironmentInterface &getEnvironment();
 
+    std::shared_ptr<SearchNode> m_root;
+    std::shared_ptr<SearchNode> tree_original_root;
```

Comment:
Is there a reason they are public? Couldn't we define getters and leave them private?